src/object.d

```diff
@@ -2922,21 +2918,53 @@ unittest
     assert(postblitRecurseOrder == order);
 }
 
+// destroy
 /********
-Destroys the given object and sets it back to its initial state. It's used to
-_destroy an object, calling its destructor or finalizer so it no longer
-references any other objects. It does $(I not) initiate a GC cycle or free
-any GC memory.
+Destroys an object by means of calling its destructor, if present. For
+types `T` that define a destructor, `destroy` then also fills the
```

Comment:
probably best to use "... that have a  destructor" here since that's the verbage used in the following "NOTE".